pym/bob/cmds/build.py

```diff
@@ -976,15 +1010,36 @@ def _cookPackageStep(self, packageStep, checkoutOnly, depth):
                     self._runShell(packageStep, "package")
                     packageHash = hashWorkspace(packageStep)
                     audit = self._generateAudit(packageStep, depth, packageHash)
+                    # copy to shared location if possible
+                    if self.__installShared and packageStep.isShared() and packageStep.isPackageRelocatable() and self.__sharedFolder:
+                        buildidhexstr =  asHexStr(packageBuildId)
+                        sharedpackagepath = os.path.join(self.__sharedFolder, buildidhexstr)
+                        print(colorize("     Install in shared location {}".format(sharedpackagepath), "32"))
+                        # create temp dir
+                        tmpdir = tempfile.mkdtemp()
+                        tmppackagepath = os.path.join(tmpdir, buildidhexstr)
+                        # copy all files from workspace - keep symlinks
+                        shutil.copytree(prettyPackagePath, tmppackagepath, symlinks=True)
+                        # copy audit
+                        shutil.copy2(os.path.join(prettyPackagePath, '..', 'audit.json.gz'), tmppackagepath)
+                        if self._checkSharedPackage(buildidhexstr):
+                            shutil.rmtree(sharedpackagepath)
+                        # move - atomic operation
+                        shutil.move(tmppackagepath, sharedpackagepath)
```

Comment:
This is *not* atomic if you cross file systems and `shutil.move` will hide it from you. Directly use `os.rename` instead...